test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperRunnerTest.cs

```diff
@@ -10,6 +10,76 @@ namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
 {
     public class TagHelperRunnerTest
     {
+        public static TheoryData TagHelperOrderData
+        {
+            get
+            {
+                // tagHelperOrders, expectedTagHelperOrders
+                return new TheoryData<int[], int[]>
+                {
+                    {
+                        new[] { 1000, int.MaxValue, 0 },
+                        new[] { 0, 1000, int.MaxValue }
+                    },
+                    {
+                        new[] { int.MaxValue, int.MaxValue, int.MinValue },
+                        new[] { int.MinValue, int.MaxValue, int.MaxValue }
+                    },
+                    {
+                        new[] { 0, 0, int.MinValue },
+                        new[] { int.MinValue, 0, 0 }
+                    },
+                    {
+                        new[] { int.MinValue, -1000, 0 },
+                        new[] { int.MinValue, -1000, 0 }
+                    },
+                    {
+                        new[] { 0, 1000, int.MaxValue },
+                        new[] { 0, 1000, int.MaxValue }
+                    },
+                    {
+                        new[] { int.MaxValue, int.MinValue, int.MaxValue, -1000, int.MaxValue, 0 },
+                        new[] { int.MinValue, -1000, 0, int.MaxValue, int.MaxValue, int.MaxValue }
+                    },
+                    {
+                        new[] { 0, 0, 0, 0 },
+                        new[] { 0, 0, 0, 0 }
+                    },
+
+                    {
+                        new[] { 1000, int.MaxValue, 0, -1000, int.MinValue },
+                        new[] { int.MinValue, -1000, 0, 1000, int.MaxValue }
+                    },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(TagHelperOrderData))]
+        public async Task RunAsync_OrdersTagHelpers(
+            int[] tagHelperOrders,
+            int[] expectedTagHelperOrders)
+        {
+            // Arrange
+            var runner = new TagHelperRunner();
+            var executionContext = new TagHelperExecutionContext("p");
+            var processOrder = new List<int>();
+            OrderedTagHelper orderedTagHelper;
```

Comment:
nit: don't bother declaring this outside the `foreach`.  use a `var` variable inside.